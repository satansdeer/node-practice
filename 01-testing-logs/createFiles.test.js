const logger = require("./logger");
jest.mock("./logger");
const dirTree = require("directory-tree");
const createFiles = require("./createFiles");

const mock = require("mock-fs");

describe("createFiles", () => {
  beforeEach(() => {
    mock({
      "./data": {}
    });
  });
  afterEach(() => {
    mock.restore();
    jest.restoreAllMocks();
  });

  it("outputs info about created files", async () => {
    await createFiles(["./data/A.txt", "./data/B.txt"]);
    // expect(logger.mock.calls).toEqual([
    //   ["Create Files\n"],
    //   ["createFile: ./data/A.txt\n"],
    //   ["createFile: ./data/B.txt\n"],
    //   ["done: ./data/A.txt\n"],
    //   ["done: ./data/B.txt\n"],
    //   ["All files created\n"]
    // ]);
    expect(logger.mock.calls).toMatchSnapshot();
  });

  it("creates specified files", async () => {
    await createFiles(["./data/A.txt", "./data/B.txt"]);
    const tree = dirTree("./data");

    expect(tree).toMatchSnapshot();
  });
});
