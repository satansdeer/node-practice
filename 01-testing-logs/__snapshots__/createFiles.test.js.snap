// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createFiles creates specified files 1`] = `
Object {
  "children": Array [
    Object {
      "extension": ".txt",
      "name": "A.txt",
      "path": "data/A.txt",
      "size": 0,
      "type": "file",
    },
    Object {
      "extension": ".txt",
      "name": "B.txt",
      "path": "data/B.txt",
      "size": 0,
      "type": "file",
    },
  ],
  "name": "data",
  "path": "./data",
  "size": 0,
  "type": "directory",
}
`;

exports[`createFiles outputs info about created files 1`] = `
Array [
  Array [
    "Create Files
",
  ],
  Array [
    "createFile: ./data/A.txt
",
  ],
  Array [
    "createFile: ./data/B.txt
",
  ],
  Array [
    "done: ./data/A.txt
",
  ],
  Array [
    "done: ./data/B.txt
",
  ],
  Array [
    "All files created
",
  ],
]
`;
